gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorage.java

```diff
@@ -200,6 +203,43 @@ void copy(
       List<String> dstObjectNames)
       throws IOException;
 
+  /**
+   * Copies metadata of the given objects. After the copy is successfully complete, each object blob
+   * is reachable by two different names. Copying between two different locations or between two
+   * different storage classes is not allowed.
+   *
+   * @param srcObjects list of the objects to copy
+   * @param dstObjects list of the objects after copy
+   * @throws java.io.FileNotFoundException if the source object or the destination bucket does not
+   *     exist
+   * @throws IOException in all other error cases
+   */
+  default void copy(List<StorageResourceId> srcObjects, List<StorageResourceId> dstObjects)
```

Comment:
Instead of 2 disjoin lists as a parameters, should we just take a Map to be less error prone and reflect intent better? Or list of `Map.Entry` objects?

Reply:
Done